os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -23,11 +23,17 @@
 #include <tinyara/config.h>
 #if defined(CONFIG_APP_BINARY_SEPARATION) && !defined(__KERNEL__)
 #include <stdio.h>
-#include <errno.h>
 #include <fcntl.h>
 #include <tinyara/fs/ioctl.h>
 #include <tinyara/mminfo.h>
 #endif
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
+#include <tinyara/watchdog.h>
+#include <fcntl.h>
+#endif
+#if defined(CONFIG_DEBUG_MM_HEAPINFO) || defined(CONFIG_APP_BINARY_SEPARATION) && !defined(__KERNEL__)
```

Comment:
I don't know why you check `defined(CONFIG_APP_BINARY_SEPARATION) && !defined(__KERNEL__)`. Could you let me know reason?

Reply:
This is just to avoid duplicate include.